File: vltdemo-java/src/main/java/com/verizon/location/platformdemo/ui/main/PageViewModel.java
package com.verizon.location.platformdemo.ui.main;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

import com.verizon.location.maps.MapMode;
import com.verizon.location.maps.VltMapOptions;

public class PageViewModel extends ViewModel {

    private MutableLiveData<Integer> mIndex = new MutableLiveData<>();
    private MutableLiveData<MapMode> mapMode = new MutableLiveData<>();
    private MutableLiveData<VltMapOptions> mapOptions = new MutableLiveData<>();
    private LiveData<String> mText = Transformations.map(mIndex, input -> "loading...");

    public void setIndex(int index) {
        mIndex.setValue(index);
    }

    public LiveData<String> getText() {
        return mText;
    }
    public void setMapMode(MapMode mode) {
        mapMode.setValue(mode);
    }
    public LiveData<MapMode> getMode() {
        return mapMode;
    }
    public void setOptions(VltMapOptions options) {
        mapOptions.setValue(options);
    }
    public LiveData<VltMapOptions> getOptions() {
        return mapOptions;
    }
}

File: vltdemo/src/main/java/com/verizon/location/platformdemo/ui/main/PageViewModel.kt
package com.verizon.location.platformdemo.ui.main

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import com.verizon.location.maps.MapMode
import com.verizon.location.maps.VltMapOptions

class PageViewModel : ViewModel() {

    private val _index = MutableLiveData<Int>()
    private val mapMode = MutableLiveData<MapMode>()
    private val mapOptions = MutableLiveData<VltMapOptions>()
    val text: LiveData<String> = Transformations.map(_index) {
        "loading..."
    }

    fun setIndex(index: Int) {
        _index.value = index
    }

    fun setMapMode(mode: MapMode?) {
        mapMode.value = mode
    }

    fun getMode(): LiveData<MapMode?>? {
        return mapMode
    }

    fun setOptions(options: VltMapOptions?) {
        mapOptions.value = options
    }

    fun getOptions(): LiveData<VltMapOptions?>? {
        return mapOptions
    }

}